gitmod/matomo.js

```diff
@@ -1,33 +1,72 @@
 addGainsight();
 var checkRequiredElementsExist = setInterval(function () {
-  if (window.gl !== 'undefined' && document.readyState == "complete" && document.querySelectorAll('[data-project]').length) {
-    clearInterval(checkRequiredElementsExist);
-    hideThings();
-    gainsightIdentify();
+    if (window.gl !== 'undefined' && document.readyState == "complete" && document.querySelectorAll('[data-project]').length) {
+      clearInterval(checkRequiredElementsExist);
+      hideThings();
+      gainsightIdentify();
+    }
+  }, 100);
+
+var oldURL = window.location.href;
+setInterval(checkURLchange, 1000);
```

Comment:
I don't really like this solution. I wouldn't want to have anything run continuously every second as this can cause slowness on the client side, especially if they have slow internet connection or are using an old browser.

I think we could use https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver to watch for changes in the DOM. DOM changes occur when I click to view an individual file from the project overview screen, so this should work for us.

Reply:
I know this is bit complex/hacky solution but I couldn't think of another approach. Will review the link you shared and see what can be improved